### PR TITLE
Add validation and docs for VirtualMachineConfigurationDTO fields

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
     @NotNull
     private BigDecimal cpu;
 
-    // Initial ram configuration, in MB.
+    @ApiModelProperty( "Initial RAM configuration, in MB" )
     @NotNull
     private Integer ram;
 
-    // In GB, Predefined disk space to be allocated as user data
+    @ApiModelProperty( "Predefined disk space to be allocated as user data, in GB" )
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: Minimum additional disk space for user data in GB

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
 import java.util.Set;
 
 public class VirtualMachineConfigurationDTO extends NamedEntityDTO
 {
+    @ApiModelProperty( "CPU cores" )
     @NotNull
     private BigDecimal cpu;
 
-    // Initial ram configuration, in MB.
+    @ApiModelProperty( "Initial RAM configuration, in MB" )
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: ~Initial~ Memory available, in MB?

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
     private BigDecimal cpu;
 
-    // Initial ram configuration, in MB.
+    @ApiModelProperty( "Memory available, in MB" )
     @NotNull
     private Integer ram;
 
-    // In GB, Predefined disk space to be allocated as user data
+    @ApiModelProperty( "Minimum additional disk space for user data, in GB" )
+    @Min( 0 )
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: Max 2048 should be reasonable

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 4]
```diff
@@ -1,24 +1,29 @@
...
 import java.math.BigDecimal;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
 public class VirtualMachineConfigurationDTO extends NamedEntityDTO
 {
+    @ApiModelProperty( "CPU cores" )
     @NotNull
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: min 1 e max 128

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 5]
```diff
@@ -1,24 +1,29 @@
...
     private BigDecimal cpu;
 
-    // Initial ram configuration, in MB.
+    @ApiModelProperty( "Memory available, in MB" )
     @NotNull
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: min 512
max 4096

File: cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java [hunk 6]
```diff
@@ -1,25 +1,34 @@
...
+    @Min( 1024 )
     private Integer ram;
 
-    // In GB, Predefined disk space to be allocated as user data
+    @ApiModelProperty( "Minimum additional disk space for user data, in GB" )
     @NotNull
+    @Max( 2048 )
```
Comment (gionn) [inline cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: Min 0?

Comment (malteo) [reply cmw-dto/src/main/java/com/liberologico/cloudesire/cmw/model/dto/VirtualMachineConfigurationDTO.java]: it's the default

**Pull request merged**